src/GitVersion.Core/Configuration/ConfigurationHelper.cs

```diff
@@ -0,0 +1,103 @@
+using GitVersion.Extensions;
+
+namespace GitVersion.Configuration;
+
+internal class ConfigurationHelper
```

Comment:
No. The ConfigurationHelper doesn't replace the ConfigurationBuilder. I'm not sure if this class is clean code. But I would describe the motivation as following:

The ConfigurationHelper is just a helper class which gets a representation of the configuration (GitVersionConfiguration class, a generic dictionary or a yaml string). It encapsulates the serialization and deserialization steps and converts from one representation to another. In advance you can define a dictionary and override the configuration property independent of the initial representation.

Reply:
A DTO does not belong in the core domain. A DTO is a way to transfer and transform data from the perimeter of the application and into one or more domain objects so the domain objects can ensure data validity and invariants. The reason I suggest it can be abstract (instead of an interface) is so we can define business logic inside the core domain and only implement the de/serialization in the infrastructural part of the application.